### ISSUE TITLE
Not working - Error: The given path's format is not supported.

### ISSUE THREAD

**Issue by beyon** [fable-elmish/templates#46]
Trying to get started with elmish but no luck with the template.

```dotnet fable yarn-start
Fable (1.2.4) daemon started on port 61225
CWD: C:\elmish\test
cmd /C yarn run start
yarn run v1.3.2
$ webpack-dev-server
Bundling for development...
Project is running at http://localhost:8080/
webpack output is served from /
Content not from webpack is served from C:\elmish\test\public
ERROR: The given path's format is not supported.
   at System.IO.Path.GetFullPath(String path)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize)
   at System.Xml.XmlDownloadManager.GetStream(Uri uri, ICredentials credentials, IWebProxy proxy, RequestCachePolicy cachePolicy)
   at System.Xml.XmlUrlResolver.GetEntity(Uri absoluteUri, String role, Type ofObjectToReturn)
   at System.Xml.XmlTextReaderImpl.FinishInitUriString()
   at System.Xml.XmlTextReaderImpl..ctor(String uriStr, XmlReaderSettings settings, XmlParserContext context, XmlResolver uriResolver)
   at System.Xml.XmlReaderSettings.CreateReader(String inputUri, XmlParserContext inputContext)
   at System.Xml.XmlReader.Create(String inputUri, XmlReaderSettings settings)
   at System.Xml.Linq.XDocument.Load(String uri, LoadOptions options)
   at Fable.CLI.ProjectCracker.getFsprojSourceFiles(String projFile)
   at Fable.CLI.ProjectCracker.crackProjects@242(FSharpList`1 acc, FSharpChoice`2 projFile)
   at Microsoft.FSharp.Collections.SeqModule.FoldBack[T,TState](FSharpFunc`2 folder, IEnumerable`1 source, TState state)
   at Fable.CLI.ProjectCracker.getProjectOptionsFromFsproj(Message msg, String projFile)
   at Fable.CLI.ProjectCracker.retry@286(FSharpChecker checker, Message msg, String projFile, DateTime retryUntil, Unit unitVar0)
   at Fable.CLI.ProjectCracker.retryGetProjectOpts(FSharpChecker checker, Message msg, String projFile)
   at Fable.CLI.ProjectCracker.getFullProjectOpts(FSharpChecker checker, Message msg, String projFile)
   at Fable.CLI.StateUtil.createProject(FSharpChecker checker, Boolean isWatchCompile, FSharpOption`1 prevProject, Message msg, String projFile)
   at Fable.CLI.StateUtil.updateState(FSharpChecker checker, FSharpMap`2 state, Message msg)
   at Fable.CLI.StateUtil.loop@195-6.Invoke(Tuple`2 _arg1)
Hash: 17658df11ca674a9b699
Version: webpack 3.6.0
Time: 8788ms
        Asset    Size  Chunks                    Chunk Names
    bundle.js  372 kB       0  [emitted]  [big]  main
bundle.js.map  424 kB       0  [emitted]         main
[./node_modules/punycode/punycode.js] ./node_modules/punycode/punycode.js 14.7 kB {0} [built]
[./node_modules/strip-ansi/index.js] ./node_modules/strip-ansi/index.js 161 bytes {0} [built]
[./node_modules/url/url.js] ./node_modules/url/url.js 23.3 kB {0} [built]
[./node_modules/url/util.js] ./node_modules/url/util.js 314 bytes {0} [built]
[./node_modules/webpack-dev-server/client/index.js?http://localhost:8080] (webpack)-dev-server/client?http://localhost:8080 7.27 kB {0} [built]
[./node_modules/webpack-dev-server/client/overlay.js] (webpack)-dev-server/client/overlay.js 3.71 kB {0} [built]
[./node_modules/webpack-dev-server/client/socket.js] (webpack)-dev-server/client/socket.js 1.04 kB {0} [built]
   [0] multi (webpack)-dev-server/client?http://localhost:8080 webpack/hot/dev-server ./src/test.fsproj 52 bytes {0} [built]
[./node_modules/webpack-dev-server/node_modules/loglevel/lib/loglevel.js] (webpack)-dev-server/node_modules/loglevel/lib/loglevel.js 6.74 kB {0} [built]
[./node_modules/webpack/hot ^\.\/log$] (webpack)/hot nonrecursive ^\.\/log$ 170 bytes {0} [built]
[./node_modules/webpack/hot/dev-server.js] (webpack)/hot/dev-server.js 1.61 kB {0} [built]
[./node_modules/webpack/hot/emitter.js] (webpack)/hot/emitter.js 77 bytes {0} [built]
[./node_modules/webpack/hot/log-apply-result.js] (webpack)/hot/log-apply-result.js 1.31 kB {0} [built]
[./node_modules/webpack/hot/log.js] (webpack)/hot/log.js 1.04 kB {0} [built]
[./src/test.fsproj] ./src/test.fsproj 217 bytes {0} [built] [failed] [1 error]
    + 75 hidden modules

ERROR in ./src/test.fsproj
Module build failed: Error: The given path's format is not supported.
    at Loader.fableUtils.client.send.then.r (C:\elmish\test\node_modules\fable-loader\index.js:67:22)
    at <anonymous>
 @ multi (webpack)-dev-server/client?http://localhost:8080 webpack/hot/dev-server ./src/test.fsproj
webpack: Failed to compile.
```

**Comment by MangelMaxime** [comment]
Could you please, give us reproduction steps ?
Like what command did you use ?

**Comment by beyon** [comment]
1) Installed template
2) dotnet new fable-elmish-react -n test --yarn
```The template "Fable.Elmish.React App" was created successfully.```
3) cd test
4) yarn install
```
yarn install v1.3.2
[1/4] Resolving packages...
[2/4] Fetching packages...
info fsevents@1.1.2: The platform "win32" is incompatible with this module.########################################################################################] 728/729
info "fsevents@1.1.2" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
[4/4] Building fresh packages...##############################################################--------------------------------------------------------------------] 843/1458
Done in 60.77s....
[2/2] ⡀ node-sass: Binary is fine
```
5) cd src
6) dotnet restore
```
C:\elmish\test\.paket\Paket.Restore.targets(86,5): warning : Detected a change ('target-framework 'netstandard2.0'') in the project file 'C:\elmish\test\src\test.fsproj', calling paket restore [C:\elmish\test\src\test.fsproj]
  Paket version 5.122.2
  Performance:
   - Disk IO: 323 milliseconds
   - Runtime: 9 seconds
  Restoring packages for C:\elmish\test\src\test.fsproj...
  Restoring packages for C:\elmish\test\src\test.fsproj...
C:\elmish\test\src\test.fsproj : warning NU1603: FSharp.Core 4.2.3 depends on System.Net.Requests (>= 4.0.11) but System.Net.Requests 4.0.11 was not found. An approximate best match of System.Net.Requests 4.3.0 was resolved.
C:\elmish\test\src\test.fsproj : warning NU1603: FSharp.Core 4.2.3 depends on System.Threading.Tasks.Parallel (>= 4.0.1) but System.Threading.Tasks.Parallel 4.0.1 was not found. An approximate best match of System.Threading.Tasks.Parallel 4.3.0 was resolved.
  Generating MSBuild file C:\elmish\test\src\obj\test.fsproj.nuget.g.props.
  Restore completed in 3,57 sec for C:\elmish\test\src\test.fsproj.
  Generating MSBuild file C:\elmish\test\src\obj\test.fsproj.nuget.g.targets.
  Restore completed in 3,62 sec for C:\elmish\test\src\test.fsproj.
```
7) dotnet fable yarn-start

**Comment by beyon** [comment]
I also tried starting a new project by first following: http://fable.io/docs/getting-started.html

And then adding Fable.Elmish and Fable.Elmish.React with paket to project.

But it also leads down to the same error when running dotnet fable yarn-start.

I'm not really that familiar with node/fable/paket having mostly used F# for desktop development, however I was up and running with Fable.Arch before and would really like to use Elmish now but it's not so easy without familiarity with the tool chain.

**Comment by MangelMaxime** [comment]
Which version of dotnet are you using ?

**Comment by beyon** [comment]
dotnet-sdk-2.0.3-win-x64

**Comment by beyon** [comment]
Tested uninstalling dotnet-sdk-2.0.3-win-x64 and installed dotnet-sdk-2.0.2-win-x64(.NET Core SDK 2.0.2 with .NET Core 2.0.0) instead.

No difference when creating a new project from the template.

**Comment by MangelMaxime** [comment]
@alfonsogarciacaro Any ideas ?

I don't access to a Windows env anymore, mine was just full of bugs and got replace a fresh linux.

**Comment by alfonsogarciacaro** [comment]
@beyon Can you please post the contents of the .fsproj file?

**Comment by beyon** [comment]
```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>netstandard2.0</TargetFramework>
  </PropertyGroup>
  <ItemGroup>
    <!-- Global to the app -->
    <Compile Include="Global.fs" />
    <!-- Info -->
    <Compile Include="Info/View.fs" />
    <!-- Counter -->
    <Compile Include="Counter/Types.fs" />
    <Compile Include="Counter/State.fs" />
    <Compile Include="Counter/View.fs" />
    <!-- Home -->
    <Compile Include="Home/Types.fs" />
    <Compile Include="Home/State.fs" />
    <Compile Include="Home/View.fs" />
    <!-- Navbar -->
    <Compile Include="Navbar/View.fs" />
    <!-- App -->
    <Compile Include="Types.fs" />
    <Compile Include="State.fs" />
    <Compile Include="App.fs" />
  </ItemGroup>
  <Import Project="..\.paket\Paket.Restore.targets" />
</Project>
````

**Comment by alfonsogarciacaro** [comment]
@beyon Thanks! Can you please check the path to Fable libraries in your system? It should be something like:

```
C:\Users\Alfonso\.nuget\packages\Fable.Elmish\1.0.0\fable\Fable.Elmish.fsproj
```

The error you reported above seems to indicate there's something wrong with this path, but I'm not sure what.

**Comment by beyon** [comment]
Well compared to your path mine has parts in lower case:
```
C:\Users\Username\.nuget\packages\fable.elmish\1.0.0\fable\Fable.Elmish.fsproj
```

**Comment by alfonsogarciacaro** [comment]
Hmm, I'm puzzled. Sorry for asking so many things, but could you please add the packages one by one to see which one causes the problem? No need to modify paket.dependencies, just start by adding Fable.Elmish to paket.references, then `dotnet restore` and `dotnet fable yarn-start`, after that try adding `Fable.Elmish.React` and so on until you get an error.

If you get the error with the first package please try removing it and adding another package to see if it's only Fable.Elmish or any package. It may be also worth to run `dotnet nuget locals all --clear` just in case.

**Comment by beyon** [comment]
Ok, not really sure why I decided to censor my user name in the path but I didn't see it as important, however it hit me that I should at leat mention I do have 'ö' in it which surprisingly in this day and age has bitten me too many times before with software that can't handle it correctly.

In fact after first having problems with creating a project from the template in my user dir I switched to under c:\elmish just to try to rule out any problem with my username path.

**Comment by alfonsogarciacaro** [comment]
Oh, then it's an easy solution, you only have to change your name 🤣 😆 🤡 

Just kidding (obviously). It could be that though I wonder why that character is not being accepted. I'm not sure if it's possible to change the place of the Nuget cache, could you investigate? You can also try to use Paket Github dependencies, but that's more cumbersome.

**Comment by et1975** [comment]
I vaguely recall seeing an issue like this with umlauts reported somewhere else for `dotnet`. You might be able to get it working by removing `storage:none` from `paket.dependencies`...maybe.

**Comment by alfonsogarciacaro** [comment]
That unfortunately won't work because Fable will still look for Elmish sources in the Nuget cache 😕

**Comment by beyon** [comment]
Following the steps from getting started with fable: http://fable.io/docs/getting-started.html

After downgrading to dotnet 2.0.2 even the fable template gives me the same error. Going back to 2.0.3 it works again.

Adding Fable.Elmish and it's broken.

If I clear the NuGet cache ```dotnet nuget locals all --clear``` dotnet restore won't work again on existing project:

```
dotnet restore
  Restoring packages for C:\elmish\FableApp2\src\FableApp2.fsproj...
  Restoring packages for C:\elmish\FableApp2\src\FableApp2.fsproj...
C:\elmish\FableApp2\src\FableApp2.fsproj : error NU1100: Unable to resolve 'dotnet-fable (>= 1.3.0)' for '.NETCoreApp,Version=v2.0'. [C:\elmish\FableApp2\FableApp2.sln]
C:\Program Files\dotnet\sdk\2.0.3\NuGet.targets(102,5): error : Value cannot be null.\r [C:\elmish\FableApp2\FableApp2.sln]
C:\Program Files\dotnet\sdk\2.0.3\NuGet.targets(102,5): error : Parameter name: path [C:\elmish\FableApp2\FableApp2.sln]
```

Trying again with new Fable project from template. Adding another semi-random package from NuGet Fable.Validation (https://www.nuget.org/packages/Fable.Validation/) also gives me the same path error.

Guess I'm stuck with trying to change NuGet package location or changing name!

Possible related but should be fixed?
https://github.com/dotnet/cli/issues/6784
https://github.com/Microsoft/msbuild/issues/2187

**Comment by MangelMaxime** [comment]
@beyon Yes, when using .Net you should not have an umlauts in your path. The tooling don't support it generally...

**Comment by beyon** [comment]
It would be good to know that it really is because of the umlaut and not something else. My attempts at changing the nuget package repository location has so far been ignored.

**Comment by beyon** [comment]
I caved in and created a user account on my computer called "elmish".
Template works so it must be the umlaut :|
Question is where does the umlaut issue originate and where to file a bug? Dotnet? Fable?

Could Fable.CLI.ProjectCracker handle things differently or is it in the System.IO implementations of dotnet? Any way to dig deeper?

**Comment by beyon** [comment]
Just to rule out there was nothing with my settings in my original account I also created a new account called "Älmish" and repeated the process. As expected it doesn't work so umlauts it is.

**Comment by MangelMaxime** [comment]
Thanks for the confirmation.

I don't know where you should report this problem but as we can't do more from the template I will close this issue. Feel free to re-open if needed.

**Comment by alfonsogarciacaro** [comment]
According to the stack, in Fable code [this is the line that fails](https://github.com/fable-compiler/Fable/blob/548b248187d269c468ce5116230d8e6cb06d70f9/src/dotnet/dotnet-fable/ProjectCracker.fs#L113) but ultimately is `System.IO.Path.GetFullPath(String path)`. Maybe you can open an issue in the [corefx repo](https://github.com/dotnet/corefx) and say loading a XML document (`System.Xml.Linq.XDocument.Load(String uri, LoadOptions options)`) fails when the path to the XML file contains an umlaut. It'll probably help to include the call stack and maybe a link to this issue :+1:

**Comment by beyon** [comment]
@alfonsogarciacaro : Can't reproduce testing umlaut paths with System.IO.Path.GetFullPath in minimal cs dotnet core project. I believe the problem is more likely to be due to the paths passed by Fable.CLI.ProjectCracker.getFsprojSourceFiles which seems to be called many times for all the refs as far as I can tell from the Fable code. Is there any way to enable more loging of Fable compilation process?

Maybe I should open an issue under the Fable project? I see no other way to find the problem without figuring out the paths used for the calls causing the error.

**Comment by alfonsogarciacaro** [comment]
@beyon It's easy to build and test a custom Fable version:

- Clone Fable repo
- Add some `printfn` statements before the failing line
- Run `build.cmd FableCLI`
- In your project, instead of `dotnet fable yarn-start`, use `dotnet ../path/to/Fable/build/fable/dotnet-fable.dll yarn-start`

You can also open an issue in Fable repo, but if I'm not able to reproduce the problem it's difficult I can do anything to fix it 😕

**Comment by beyon** [comment]
@alfonsogarciacaro Thanks for the steps, I added a printfn to see the path causing the error, it's one of two cases of problems with paths containing characters like 'ö','ä', etc currently with .net core on my machine - a path using forward slashes and troublesome characters. The other problem case I found was relative paths with same characters.

I can hack fable to work with my user account but in the future the issues should hopefully be fixed in .net core.
